src/player_activity.cpp

```diff
@@ -242,11 +243,11 @@ void player_activity::do_turn( player *p )
             break;
         default:
             // Based on speed, not time
-            if( p->moves <= moves_left ) {
-                moves_left -= p->moves;
+            if( p->moves * season_mult <= moves_left ) {
```

Comment:
It's a comparison, not an assignment.

Now that you mention it, though... does C++ still fall victim to that odd behaviour where 8.0 != 8, because 8.0 is _technically_ 7.99999 or 8.000001?